java-surefire/src/main/java/org/sonar/plugins/surefire/api/SurefireUtils.java

```diff
@@ -47,27 +61,61 @@ private SurefireUtils() {
    * @param pathResolver Path solver.
    * @return The directory containing the surefire reports or default one (target/surefire-reports) if not found (not configured or not found).
    */
-  public static File getReportsDirectory(Settings settings, FileSystem fs, PathResolver pathResolver) {
+  public static List<File> getReportsDirectories(Settings settings, FileSystem fs, PathResolver pathResolver) {
     File dir = getReportsDirectoryFromProperty(settings, fs, pathResolver);
+    List<File> dirs = getReportsDirectoriesFromProperty(settings, fs, pathResolver);
+    if (dir != null ^ dirs != null) {
+      // only one of both properties is set
+      if (dirs != null) {
+        return dirs;
+      }
+      return Collections.singletonList(dir);
+    }
     if (dir == null) {
-      dir = new File(fs.baseDir(), "target/surefire-reports");
+      // both properties are not set
+      return Collections.singletonList(new File(fs.baseDir(), "target/surefire-reports"));
     }
-    return dir;
+    // both properties are set
+    LOGGER.info("Property '" + SUREFIRE_REPORTS_PATH_PROPERTY + "' is deprecated and will be ignorded, as property '" + SUREFIRE_REPORT_PATHS_PROPERTY + "' is also set.");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "SUREFIRE_REPORTS_PATH_PROPERTY"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)